src/lib.rs

```diff
@@ -423,26 +258,26 @@ impl fmt::Display for Display<'_> {
         // We overestimate how many spaces we need, but this is not
         // part of the loop and it's therefore not super important to
         // get exactly right.
-        let padding = " ".repeat(self.grid.widest_cell_length);
+        let padding = " ".repeat(self.widest_cell_width);
 
         for y in 0..self.dimensions.num_lines {
             for x in 0..self.dimensions.widths.len() {
-                let num = match self.grid.options.direction {
+                let num = match self.options.direction {
                     Direction::LeftToRight => y * self.dimensions.widths.len() + x,
                     Direction::TopToBottom => y + self.dimensions.num_lines * x,
                 };
 
                 // Abandon a line mid-way through if that’s where the cells end
-                if num >= self.grid.cells.len() {
+                if num >= self.cells.len() {
                     continue;
                 }
 
-                let cell = &self.grid.cells[num];
-                let contents = &cell.contents;
+                let contents = &self.cells[num];
+                let width = self.widths[num];
                 let last_in_row = x == self.dimensions.widths.len() - 1;
 
                 let col_width = self.dimensions.widths[x];
-                let padding_size = col_width - cell.width;
+                let padding_size = col_width - width;
```

Comment:
This line can lead to a panic with "attempt to subtract with overflow". For example, if you set `width: 15` in the basic example.

Reply:
Should be fixed now! I set the width for the fallback to a single column incorrectly. Nice find!

Reply:
Right? damn I was gonna say... good catch lol

Reply:
@tertsdiepraam I played a bit with the basic example and wanted to see what happens if the width is low/zero ;-)